Applications/Resnet/jni/main.cpp

```diff
@@ -15,6 +15,70 @@
 #include <vector>
 
 #include <layer.h>
+#include <model.h>
+
+using LayerHandle = std::shared_ptr<ml::train::Layer>;
+using ModelHandle = std::unique_ptr<ml::train::Model>;
+
+/**
+ * @brief make "key=value" from key and value
+ *
+ * @tparam T type of a value
+ * @param key key
+ * @param value value
+ * @return std::string with "key=value"
+ */
+template <typename T>
+static std::string withKey(const std::string &key, const T &value) {
+  std::stringstream ss;
+  ss << "key=" << value;
+  return ss.str();
+}
+
+template <typename T>
+static std::string withKey(const std::string &key,
+                           const std::vector<T> &value) {
+  if (value.empty()) {
+    throw std::invalid_argument("empty vector cannot be converted");
+  }
+
+  std::stringstream ss;
+  ss << "key=";
+  for (unsigned int i = 0; i < value.size() - 1; ++i) {
+    ss << value.at(i) << ',';
+  }
+  ss << value.back();
+
+  return ss.str();
+}
+
+/**
+ * @brief resnet block
+ *
+ * @param block_name name of the block
+ * @param input_name name of the input
+ * @param filters number of filters
+ * @param kernel_size number of kernel_size
+ * @param downsample downsample to make output size 0
+ * @return std::vector<LayerHandle> vectors of layers
+ */
+std::vector<LayerHandle> resnetBlock(const std::string &block_name,
```

Comment:
are you planning to create the model with the API?

Reply:
Not directly, but this is the primitive block of a resnet

Reply:
As you are using c++ API functions in the next PR, so I will take it as a yes, using API for model creation (compared to using ini to create the model).

Reply:
Thanks, I misunderstood the question, thought the question was related to the line